src/scalar_impl.h

```diff
@@ -293,16 +304,161 @@ static void secp256k1_scalar_split_lambda(secp256k1_scalar *r1, secp256k1_scalar
  * The derivation is described in the paper "Efficient Software Implementation of Public-Key
  * Cryptography on Sensor Networks Using the MSP430X Microcontroller" (Gouvea, Oliveira, Lopez),
  * Section 4.3 (here we use a somewhat higher-precision estimate):
- * d = a1*b2 - b1*a2
- * g1 = round((2^272)*b2/d)
- * g2 = round((2^272)*b1/d)
+ * d = a1*b2 - b1*a
+ * g1 = round(2^384 * b2/d)
+ * g2 = round(2^384 * (-b1)/d)
+ *
+ * (Note that 'd' is also equal to the curve order, n, here because [a1,b1] and [a2,b2]
+ * can be found as outputs of the Extended Euclidean Algorithm on inputs 'n' and 'lambda').
+ *
+ * The function below splits a in r1 and r2, such that
```

Comment:
In general, `a` and  (`a1`,`a2`) are confusing because they're not related. Maybe we can just rename `a1`,`a2`? They're used only in comments, so that should be easy.

Reply:
How do we feel about renaming the input parameter as `k`?

Reply:
Totally fine, that matches the papers. I just didn't suggest this one because it's more work.

Reply:
And rename it in the header file, `scalar.h` too?

Reply:
Yes, consistency with the header is a good thing IMO.

Reply:
I think there are a few places in this comment where it's still `a`.